@xen-orchestra/xapi/vdi.mjs

```diff
@@ -134,7 +134,16 @@ class Vdi {
   }
 
   async disconnectFromControlDomain(vdiRef) {
-    const vbdRefs = await this.getField('VDI', vdiRef, 'VBDs')
+    let vbdRefs
+    try {
+      vbdRefs = await this.getField('VDI', vdiRef, 'VBDs')
+    } catch (err) {
+      // since we can't get the info of this record we assume it is delete or deleting
+      if (err.code === 'HANDLE_INVALID') {
```

Comment:
It does not really make sense for me how this could happen so we should at least add a comment :)